operator/internal/status/lokistack.go

```diff
@@ -149,53 +156,16 @@ func checkForZoneawareNodes(ctx context.Context, k client.Client, zones []lokiv1
 	return true, true, nil
 }
 
-func updateCondition(ctx context.Context, k k8s.Client, req ctrl.Request, condition metav1.Condition) error {
-	var stack lokiv1.LokiStack
-	if err := k.Get(ctx, req.NamespacedName, &stack); err != nil {
-		if apierrors.IsNotFound(err) {
-			return nil
-		}
-		return kverrors.Wrap(err, "failed to lookup LokiStack", "name", req.NamespacedName)
-	}
+func generateWarnings(schemas []lokiv1.ObjectStorageSchema) []metav1.Condition {
+	warnings := make([]metav1.Condition, 0, 2)
```

Comment:
```suggestion
	warnings := make([]metav1.Condition, 0, 1)
```

nit: We allocate only one warning, the backing array of this slice can stay with a capacity of 1. Or do I miss something, like a second warning condition somewhere else?

Reply:
This one is just a typo I guess

Reply:
The two was there on purpose, to have room for the condition that is always there (ready, pending, failed, ...) and one warning.